vm/actor/src/util/set_multimap.rs

```diff
@@ -97,7 +101,8 @@ where
                 .map_err(|e| format!("Could not parse key: {:?}, ({})", &k.0, e))?;
 
             // Run function on all parsed keys
-            f(v)
+            f(v).map_err(|e| e.to_string())
```

Comment:
This might be misleading, you are requiring the generic function error is ActorError, then converting to string. If that exit code doesn't matter/isn't used, then the generic function error should remain a string

Reply:
The error return type here is `String` as its within [set.rs for_each](https://github.com/ChainSafe/forest/blob/dustin/market-updates/vm/actor/src/util/set.rs#L61) function. I still want to return an ActorError for set_multimap for_each.

Reply:
So yeah there is those exit codes that are lost and converted to `ErrIllegalState`, this error cannot be lost in the conversion to string (I'll try to think of a clean way around this)

Reply:
I made some changes here https://github.com/ChainSafe/forest/commit/f5dd90c3d646d570ab7948dc6ae02286ffd3a552 To match go impl, I think the cleanest way is to downcast the error like this. I think it matches the exit codes 1:1 but please verify. 
Keep in mind when they call rt.Abort with an exit code, we need that code to be kept and eventually returned. Please ask any questions about this or offer alternatives, I don't want this to silently go in, but feel free to cherry pick the commit if you agree

Reply:
Made requested changes but as discussed removed the actor error from set_multimap.